### PR TITLE
UnexpecTEDIntentPolicy: Update tests

### PR DESCRIPTION
**Pull request by dakshvar22**
**Proposed changes**:
- Fix for https://github.com/RasaHQ/rasa/pull/8913#discussion_r665244346

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Review by alwx — APPROVED** [review]

File: tests/core/test_test.py [hunk 1]
```diff
@@ -129,16 +148,21 @@ async def test_action_unlikely_intent_1(
         """
     )
 
+    # We train on the above story so that RulePolicy can memorize
+    # it and we don't have to worry about other actions being
+    # predicted correctly.
+    agent = await _train_rule_based_agent(moodbot_domain, file_name)
```
Comment (wochinge) [inline tests/core/test_test.py:154]: Out of curiosity: What didn't work with the monkey patching?

File: tests/core/test_test.py [hunk 2]
```diff
@@ -103,8 +104,26 @@ async def test_testing_valid_with_non_e2e_core_model(core_agent: Agent):
...
+    agent.train(training_data)
+
+    return agent
+
+
 async def test_action_unlikely_intent_1(
-    monkeypatch: MonkeyPatch, tmp_path: Path, unexpected_intent_policy_agent: Agent
+    monkeypatch: MonkeyPatch, tmp_path: Path, moodbot_domain: Domain
 ):
     monkeypatch.setattr(
```
Comment (wochinge) [inline tests/core/test_test.py:128]: can we delete the monkey patching then?

Comment (wochinge) [reply tests/core/test_test.py:128]: (same for the other occasions?)

File: tests/core/test_test.py (hunk 1 above)
Comment (dakshvar22) [reply tests/core/test_test.py:154]: The monkey patching is for triggering `action_unlikely_intent` at a specific instance. All other actions still need to be predicted correctly which is what this rule based agent would be trained to do.

Comment (wochinge) [reply tests/core/test_test.py:154]: ooooh, I see 👍🏻

Comment (dakshvar22) [reply tests/core/test_test.py:154]: @wochinge Do you have any other comments on this or can I merge it?

Comment (wochinge) [reply tests/core/test_test.py:154]: Sorry, I was waiting for an answer to the second comment but just had another look and makes sense 👍🏻

Comment (dakshvar22) [reply tests/core/test_test.py:154]: Ahh, my bad too. I could have updated that one too. 🙈

**Pull request merged**